File: source/dotnet/source/cmf.bus/cmf.eventing/IEventHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using cmf.bus;


namespace cmf.eventing
{
    public interface IEventHandler
    {
        string Topic { get; }


        void Handle(object ev, IDictionary<string, string> headers);

        void HandleFailed(Envelope env, Exception ex);
    }
}

File: java/src/main/java/cmf/eventing/IEventHandler.java
package cmf.eventing;

import java.util.Map;

import cmf.bus.Envelope;

public interface IEventHandler<TEVENT> {

    Class<TEVENT> getEventType();

    Object handle(TEVENT event, Map<String, String> headers);

    Object handleFailed(Envelope envelope, Exception e);
}
